### PR TITLE
Breadcrumbs - second component migration to Material-UI.

### PR DESCRIPTION
**Pull request by vincelugli**
Migrates breadcrumb component from Bootstrap to Material-UI. Moved styles to the theme to be reused across the app.

Additionally, this updates the tests to work with `withStyles()`. Going to open up an issue to improve how our tests work; I think there's a lot we can do to make them better. Current fix is to start using [enzyme](https://github.com/airbnb/enzyme), by AirBnB.

breadcrumb-migration

### CODE REVIEW

File: src/components/RouterBreadcrumbs.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import Breadcrumbs from '@material-ui/lab/Breadcrumbs';
+import Link from '@material-ui/core/Link';
+import Typography from '@material-ui/core/Typography';
+import { Link as RouterLink } from 'react-router-dom';
+import { Paper, withStyles } from '@material-ui/core';
+
+const styles = (theme) => ({
+  paper: {
+    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 2}px`,
```
Comment (shadanan) [inline src/components/RouterBreadcrumbs.js]: Please for comment why we do `* 2` here.

File: src/components/RouterBreadcrumbs.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ */
+const RouterBreadcrumbs = ({ pathname, classes }) => {
+  const breadcrumbItems = [];
+  const viewNames = ['browser', 'viewer', 'editor', 'repos'];
+  const keyBase = 'breadcrumb-item-';
+
+  const subDirectoriesArr = pathname.split('/');
+
+  // pathname starts with '/', so discard. Pathname will sometimes end with '/', so discard that as well.
+  subDirectoriesArr.shift();
```
Comment (shadanan) [inline src/components/RouterBreadcrumbs.js]: Sometimes due to weird situations, you might end up with `//` in the URI. You might want to generalize this as a filter that just filters out all `''`:

    subDirectoriesArr = subDirectoriesArr.filter((value) => value !== '');

Maybe you can even do `!!value`.

File: src/components/RouterBreadcrumbs.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    subDirectoriesArr.pop();
+  }
+
+
+  let currDir = '';
+  for (let i = 0; i < subDirectoriesArr.length; i++) {
+    const directory = subDirectoriesArr[i];
+    currDir = currDir.concat('/', directory);
+
+    if (viewNames.indexOf(directory) !== -1) {
```
Comment (shadanan) [inline src/components/RouterBreadcrumbs.js]: This introduces a bug where breadcrumbs doesn't work correctly if the user has a folder in their repository with any of browser, viewer, editor, repos.

File: src/components/RouterBreadcrumbs.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ * Returns a list of breadcrumbs based on current path of navigation
+ * @param {string} pathname current URI pathname
+ * @return {Array} List of BreadcrumbItems
+ */
+const RouterBreadcrumbs = ({ pathname, classes }) => {
+  const breadcrumbItems = [];
+  const viewNames = ['browser', 'viewer', 'editor', 'repos'];
+  const keyBase = 'breadcrumb-item-';
+
+  const subDirectoriesArr = pathname.split('/');
```
Comment (shadanan) [inline src/components/RouterBreadcrumbs.js]: I recommend extracting this interesting business logic into a function, and then writing tests for it in the spec.js file.

File: src/components/RouterBreadcrumbs.js (hunk 1 above)
Comment (vincelugli) [reply src/components/RouterBreadcrumbs.js]: Removing this actually. Took it from the docs on `Breadcrumbs`, but isn't really necessary.

File: src/components/RouterBreadcrumbs.js (hunk 3 above)
Comment (vincelugli) [reply src/components/RouterBreadcrumbs.js]: 👍 changing this to index based, since our URL shouldn't be changing.

**Pull request merged**